### PR TITLE
Rc0.2.1

### PR DESCRIPTION
**Pull request by bollig**
- Cleaned up cloudformation Parameters by adding ParameterGroups. Unsightly Assets are wrangled to "Other settings" at the bottom of the page. 
- Expanded options for FSx Lustre to modify storage capacity, PERSISTENT vs SCRATCH, and ImportPolicy
- Added Default values for Assets::ArtifactHash (AWS::NoValue) so those parameters never need to be specified. 
- Added an example inherit.yaml showing how to nest notears within another cloudformation template. Intend to use this approach to extend with custom post_install_script for workload-specific software.

### CODE REVIEW

File: inherit.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Resources:
+    NestedStack:
+      Type: 'AWS::CloudFormation::Stack'
+      Properties: 
+          TemplateURL: 'https://notearshpc-quickstart.s3.amazonaws.com/0.2.1/cfn.yaml'
```
Comment (sean-smith) [inline inherit.yaml:6]: Super cool! let's add this as an example to the README as well

File: pcluster/pcluster_stack.py [hunk 2]
```diff
@@ -31,14 +31,25 @@ def get_version_list(package_name):
     versions = data["releases"].keys()
     return list(versions)
 
+#def get_git_version_list(git_owner, package_name):
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py]: remove this

File: pcluster/pcluster_stack.py [hunk 3]
```diff
@@ -52,7 +63,10 @@ def __init__(self, scope: cdk.Construct, id: str, **kwargs) -> None:
         # S3 URI for Config file
         config = cdk.CfnParameter(self, 'ConfigS3URI', description='Set a custom parallelcluster config file.', default='https://notearshpc-quickstart.s3.amazonaws.com/{0}/config.ini'.format(__version__))
 
-        spack_version = cdk.CfnParameter(self, 'SpackVersion', description='Specify a custom Spack version. See https://github.com/spack/spack/releases for options.', default='v0.16.0', type='String', allowed_values=get_git_version_list('spack','spack'))
+        #spack_version = cdk.CfnParameter(self, 'SpackVersion', description='Specify a custom Spack version. See https://github.com/spack/spack/releases for options.', default='v0.16.0', type='String', allowed_values=get_git_version_list('spack','spack'))
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py]: remove this

File: pcluster/pcluster_stack.py [hunk 4]
```diff
@@ -108,20 +122,36 @@ def __init__(self, scope: cdk.Construct, id: str, **kwargs) -> None:
...
+        lustre_storage_capacity = cdk.CfnParameter(self, 'FSxLustreStorageCapacity', description='FSx Lustre filesystem capacity (in GiB). Scratch and persistent SSD-based file systems can be created in sizes of 1200 GiB or in increments of 2400 GiB. Persistent HDD-based file systems with 12 MB/s and 40 MB/s of throughput per TiB can be created in increments of 6000 GiB and 1800 GiB, respectively.', default=1200, type='Number')
+
+
+        fsx_lustre_config = {
+            "autoImportPolicy": lustre_import_policy.value_as_string,
+            "deploymentType": lustre_type.value_as_string,
+            "exportPath": 's3://%s' % ( data_bucket.bucket_name ),
+            "importPath": 's3://%s' % ( data_bucket.bucket_name ),
+            "perUnitStorageThroughput": lustre_performance.value_as_number,
+            #"driveCacheType": cdk.Fn.condition_if(drive_cache_is_none.logical_id, cdk.Aws.NO_VALUE, lustre_drive_cache.value_as_string),
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py]: remove this

File: pcluster/pcluster_stack.py [hunk 5]
```diff
@@ -288,7 +318,7 @@ def repl(mykey, myr, myval):
         pcluster_post_install_script.grant_read(cloud9_role)
         pcluster_config_script.grant_read(cloud9_role)
 
-        create_user = cdk.CfnParameter(self, "CreateUserAndGroups", default="false", type="String", allowed_values=['true','false'])
+        create_user = cdk.CfnParameter(self, "CreateUserAndGroups", default="false", type="String", allowed_values=['true','false'], description='Provision an additional IAM user (Researcher) account and Admin/PowerUser groups to simplify onboarding multiple users in one account.')
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py]: Add to the description: 

Note: this requires IAM permissions to create a user and group.

File: pcluster/pcluster_stack.py [hunk 6]
```diff
@@ -317,8 +347,8 @@ def repl(mykey, myr, myval):
         cdk.CfnOutput(self, 'UserLoginUrl', value="".join(["https://", self.account,".signin.aws.amazon.com/console"]), condition=user_condition)
         cdk.CfnOutput(self, 'UserName', value=user.ref, condition=user_condition )
 
-        base_os = cdk.CfnParameter(self, "Operating System", default="alinux2", type="String", allowed_values=['ubuntu1804','alinux2']).value_as_string
-        cdk.CfnOutput(self, 'OS', value=base_os)
+        base_os = cdk.CfnParameter(self, "Operating System", default="alinux2", type="String", allowed_values=['ubuntu1804','alinux2'])
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py]: Any issues with centos? ;)

File: pcluster/pcluster_stack.py [hunk 7]
```diff
@@ -486,7 +517,58 @@ def repl(mykey, myr, myval):
...
-        overall_budget.cfn_options.condition = cdk.CfnCondition(self, "BudgetCondition", expression=cdk.Fn.condition_equals(create_budget, "true"))
+        overall_budget.cfn_options.condition = cdk.CfnCondition(self, "BudgetCondition", expression=cdk.Fn.condition_equals(create_budget.value_as_string, "true"))
+
+
+
+        self.template_options.metadata = {
+            'AWS::CloudFormation::Interface': {
+                'ParameterGroups': [
+                    {
+                        'Label': { 'default': 'ParallelCluster Configuration' },
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py:516]: This is really great

File: pcluster/pcluster_stack.py [hunk 8]
```diff
@@ -31,14 +31,25 @@ def get_version_list(package_name):
...
     import json
-    import requests
-    from distutils.version import StrictVersion
+    import os
+    from github import Github
 
-    url = "https://api.github.com/repos/%s/%s/tags" % (git_owner, package_name,)
-    data = requests.get(url).json()
-    versions = [v['name'] for v in data]
+    g = Github(os.environ.get('GIT_AUTH_TOKEN'))
```
Comment (sean-smith) [inline pcluster/pcluster_stack.py:40]: Where does `GIT_AUTH_TOKEN` come from?

File: pcluster/pcluster_stack.py (hunk 6 above)
Comment (bollig) [reply pcluster/pcluster_stack.py]: My frugality? Per-second billing! 

Ok, ok. I'll put the other OS options in there. Use at your own risk

File: pcluster/pcluster_stack.py (hunk 8 above)
Comment (bollig) [reply pcluster/pcluster_stack.py:40]: Your github account. Settings -> Developer Settings -> Personal Access Tokens. (Dont check any boxes). Put it in your bashrc as
```
export GIT_AUTH_TOKEN=XXXXXXX
```

File: inherit.yaml (hunk 1 above)
Comment (bollig) [reply inherit.yaml:6]: done.

File: pcluster/pcluster_stack.py (hunk 8 above)
Comment (bollig) [reply pcluster/pcluster_stack.py:40]: Added doc to README for developers

File: pcluster/pcluster_stack.py (hunk 6 above)
Comment (bollig) [reply pcluster/pcluster_stack.py]: Done.

File: pcluster/pcluster_stack.py (hunk 5 above)
Comment (bollig) [reply pcluster/pcluster_stack.py]: Done

File: pcluster/pcluster_stack.py (hunk 4 above)
Comment (bollig) [reply pcluster/pcluster_stack.py]: done.

File: pcluster/pcluster_stack.py (hunk 3 above)
Comment (bollig) [reply pcluster/pcluster_stack.py]: Done. And added 'develop' to the version options

File: pcluster/pcluster_stack.py (hunk 2 above)
Comment (bollig) [reply pcluster/pcluster_stack.py]: done.

**Pull request merged**